source/lac/petsc_vector_base.cc

```diff
@@ -210,6 +210,27 @@ namespace PETScWrappers
 
 
 
+  void
+  VectorBase::clear ()
+  {
+    if (attained_ownership)
+      {
+#if DEAL_II_PETSC_VERSION_LT(3,2,0)
+        const int ierr = VecDestroy (vector);
+#else
+        const int ierr = VecDestroy (&vector);
```

Comment:
We can initialize <code>vector</code> to an empty sequential vector as in derived classes. This brings us to your next comment. Isn't it cleaner to have this function also in the base class? But no problem, you decide.